### DISCUSSION TITLE
Instanciamento de TJsonObject com Req.Body

### DISCUSSION THREAD

**Discussion by valdirdill** [HashLoad/horse#385] [General]
Bom dia,

Estou com uma dúvida em relação ao instanciamento do object que chega no req.body.

O que ocorre é que quando executo isto: _VMeuJsonObject := Req.Body<TJSONObject>_, às vezes, é acionada a "_function THorseRequest.Body: string_", e outras vezes é acionada a "_function THorseRequest.Body(const ABody: TObject): THorseRequest_".
Quando acontece a primeira situação, o VMeuJsonObject permanece nil. Já na segunda, o json fica certinho.
Pelo que vi, no primeiro caso, ele (o THorseRequest) entende que o que está vindo é uma string simples no req.body e não um object.

Mas verifiquei e, em ambas as situações, sempre vem um json exatamente igual. Assim -> {"operador":"1", "nome":"teste", "status":"10"}
Tanto é que se, aos invés de usar isto -> _Req.Body<TJSONObject>_, eu fizer parse via código manual, o VMeuJsonObject é populado corretamente com os valores.

Se puderem me dar alguma dica...

Obrigado!

**Comment by viniciussanchez** [comment]
O padrão é que o Req.Body seja string. 
Se você utilizar o middleware Jhonson, e quem efetuar a requisição, enviar um JSON + um Header Content-Type como application/json, o middleware Jhonson, vai identificar que o conteúdo da requisição é um JSON, e vai criar um Objeto, deixando ele disponível em Req.Body<TJSONObject> por exemplo.

**Comment by valdirdill** [comment]
Entendi.
Obrigado!
